### ISSUE TITLE
ReverseProxyResource doesn't recursively pass its reactor setting

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4403]
|<img alt="aschmolck's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| aschmolck reported|
|-|-|
|Trac ID|trac#4403|
|Type|defect|
|Created|2010-04-16 08:00:45Z|

ReverseProxyResource takes an optional reactor argument in its constructor, which is not passed when a new instance of itself is created in its getChild method.

Attachments:

* [my-twisted-patch.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7fa/7fa2fbb12053cadf46b52f1ca087e43c97aa4be3/d5bf14bfe4ce8fc7b9256a069dbe396aeaef28e5.patch) (1578 bytes) - added by aschmolck on 2010-04-16 08:43:05Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4403 4403
type__defect defect
reporter__aschmolck aschmolck
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1271404845000000 1271404845000000
changetime__1271427028000000 1271427028000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks!  There are a couple minor coding standard issues with the patch, but it basically looks good.  Here are the trivial changes I'm going to make before I apply it:

  * Update copyright date on proxy.py
  * Use `assertIdentical` instead of `failIf` for the new assertion in `test_getChild`
  * Preserve the two blank lines after `test_getChild` before the next test method
  * Update the `test_getChild` docstring to mention that the reactor value is to be passed on as well.
  * Add a (misc) news fragment

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set status to `closed`|
|-|-|

(In [28824]) Apply patch to propagate reactor from a ReverseProxyResource to its children

Author: aschmolck
Reviewer: exarkun
Fixes: [#4403](https://github.com/twisted/twisted/issues/4403)

When creating a new ReverseProxyResource in ReverseProxyResource.getChild, pass
on the reactor the parent was created with to the new child.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
